### ISSUE TITLE
No way to disallow entering letters into the input

### ISSUE THREAD

**Issue by adamhinckley** [jackocnr/intl-tel-input#1540]
### Steps to reproduce
1.  Select US as the country and enter  `208-for-rent` in the phone input
2.  When I use `getNumber` it accurately shows that the letters have been converted to numbers that correspond to the numbers on a phone dial pad.

### Expected behaviour
I would like a way to disallow entering letters into the input.  Does that exist somewhere in the docs and I just missed it?

### Actual behaviour
Letters can be used as valid input.

### Initialisation options
```
const initOptions = {
	initialCountry: countryRef.current,
	utilsScript: utilScript
};
```

**Comment by jackocnr** [comment]
Thanks for raising this. I'd appreciate your thoughts on the subject: can you describe the reason(s) why you think someone might type a letter in the input in the first place (is it just by accident or are there other possible reasons?)? And why you think silently ignoring the user's action would be better than allowing it and then if they've typed the equivalent of a valid number (like in your example) then converting the letters to numbers, else failing validation?

**Comment by adamhinckley** [comment]
Thanks for the reply.

I don't know why someone would intentionally type a number in a phone input in my case, but I can see it happening by accident then being seen as valid when it's not the number the user intended to enter.  I also see why it could be desired in some cases outside of my own.

Rather than silently ignoring the letter, I think it would be ideal for it to be seen as invalid so I can show an error to the user.  Is this something that could be added as a boolean to the config?

Or Is there a way to extract the string, as typed by the user, so I can run my own validation on it because `getNumber()` returns the value with the letters replaced with numbers?  If that can be done I can check for the existence of letters on my own.

**Comment by adamhinckley** [comment]
We ended up finding a work around.

Context: this is in a React app and we used the React component from this library (thank you 🙏) as a guide to build our own React component.

This is how we are able to get the input as the user typed and check for letters
```
const num = itiRef.current.getNumber();

// remove non-english letters and accents
const rawValue = inputEl.value.normalize('NFD').replace(/[\u0300-\u036f]/g, '');

const isValid = itiRef.current.isValidNumberPrecise() && !/[A-Za-z]/.test(rawValue);
```

**Comment by jackocnr** [comment]
> Rather than silently ignoring the letter, I think it would be ideal for it to be seen as invalid so I can show an error to the user

I agree with this. I think we should add an extra condition to isValidNumber and isValidNumberPrecise to confirm there are no alpha chars in there. We use libphonenumber for validation and I think they support alpha char substitutions because in some cases that is useful to them e.g. for when users want to dial numbers in that way, but I don't think it would ever be useful for this plugin. I'll re-open this issue to track this feature. I would be open to a pull request that adds this extra check in those two methods.

**Comment by jackocnr** [comment]
Added in v20.0.0

**Comment by jackocnr** [comment]
> we used the React component from this library (thank you 🙏) as a guide to build our own React component

@adamhinckley was there a particular reason the provided component didn't fulfil your needs? Or you just preferred a different component/prop architecture or something? I would welcome any feedback on how to improve it. Cheers

**Comment by jackocnr** [comment]
In addition to the previous work, I have now added a `strictMode` option, which prevents users from entering invalid characters. Added in v20.2.0.

**Comment by mkamranashraf** [comment]
I can still paste invalid characters in the input despite strictMode on. Am I missing some configuration?

**Comment by Glideh** [comment]
That's right. Is it on purpose ?
This `strictMode` is useful but it'd be even more for me if it prevented the paste too

**Comment by jackocnr** [comment]
What would you expect to happen when the user tries to paste something containing both numbers and letters? Should it just ignore the paste altogether? I suppose the alternative would be to strip the invalid characters, but I think that would be confusing.

What if they try to paste an already formatted number, e.g. "+1 (702) 418-1234" - then it seems wrong to just ignore that completely. I suppose we could check if the string consists entirely of numerics and valid phone formatting chars (non-alpha), and if it does, then we strip the formatting chars, put the valid chars in the input and then apply our own formatting? I'm not convinced... that also might feel weird/unexpected. What do you guys think?

**Comment by Glideh** [comment]
I was wondering too. I think I would not be surprised if an invalid paste were simply ignored in `strictMode`.
Another idea would be to simulate the manual entry with the pasted string, so it would be showing +1 702-418-1234 at the end.
And "abc123" pasted would end with "123"

**Comment by jackocnr** [comment]
Sounds good to me. I'll reopen this issue to manage this new feature. I don't have time to implement this myself right now, but I'd welcome a pull request that adds this functionality. We'll need to add a new "paste" event handler (only if strictMode is enabled). And I agree we should simply strip all invalid chars, and format the inserted chars in the usual way. Cheers.

Additional notes:
- if separateDialCode is enabled, then don't allow a leading plus in the pasted value
- if existing input value has a leading plus (which isn't currently selected and so won't be replaced by the pasted value) then don't allow leading plus in the pasted value
- We'll need to apply the usual cap (length) to the new input value
- We'll need to make sure the pasted value is formatted correctly (maybe this will happen automatically by the "input" event handler)

**Comment by jackocnr** [comment]
Released in v25.10.7. Do test it and let me know if you find any issues @mkamranashraf @Glideh

**Comment by Glideh** [comment]
Works fine for me, thx

**Comment by Glideh** [comment]
Ha the paste still seems to allow more chars than allowed

**Comment by jackocnr** [comment]
> the paste still seems to allow more chars than allowed

I can't reproduce this on https://intl-tel-input.com/examples/strict-mode.html

Can you give clear steps to reproduce?

**Comment by Glideh** [comment]
On Linux, I copy `01234567890123456789` with a double click and paste with a middle click (same behavior with an explicit right click contextual menu copy/paste).
Browser used: Brave (Chrome based)

**Comment by jackocnr** [comment]
Thanks for reporting this. It was due to a quirk in libphonenumber. It should now be fixed in v25.10.8.
